src/main/java/sign_up/use_case/SignUpInteractor.java

```diff
@@ -0,0 +1,80 @@
+package sign_up.use_case;
+
+import entities.CurrentUser;
+import entities.User;
+import sign_up.use_case.exceptions.MakeUserException;
+
+public class SignUpInteractor implements SignUpInputBoundary{
+    // An object that accesses the database
+    private final SignUpDataAccessInterface dataAccess;
+
+    // An object that accesses the UI
+    private final SignUpOutputBoundary outputBoundary;
+
+    // An object whose job is to create User elements
+    private final UserFactory userFactory;
+
+    public SignUpInteractor(SignUpDataAccessInterface dataAccess, SignUpOutputBoundary outputBoundary){
+        this.dataAccess = dataAccess;
+        this.outputBoundary = outputBoundary;
+        this.userFactory = new UserFactory();
+    }
+
+    @Override
+    public void signUp(SignUpRequestModel requestModel) {
+        try {
+            SignUpResponseModel responseModel = this.signUpHelper(requestModel);
+            this.outputBoundary.present(responseModel);
+        }
+        catch (MakeUserException e) {
+            SignUpResponseModel responseModel = new SignUpResponseModel(false, e.getMessage());
+        }
```

Comment:
shouldn't there be a 'this.outputBoundary.present(responseModel) here as well? in order to present the response with the error message

Reply:
Yes good catch, thank you!